File: Assets/Scripts/ButtonEndless.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

// The button that starts endless mode.
public class ButtonEndless : MonoBehaviour
{
    void Start()
    {
        // Subscribe to button clicks.
        EventBus.Subscribe<ButtonPress>(_OnClick);
    }

    // When receiving a click event for this button, load endless.
    void _OnClick(ButtonPress e)
    {
        if (e.selected_button == gameObject) UnityEngine.SceneManagement.SceneManager.LoadScene("Level10");
    }
}

File: component_types/ButtonEndless.lua
ButtonEndless = {

    focus = nil,

    _OnPress = function (self, event)
        if event.row == self.focus.row and event.column == self.focus.column then
            Scene.Load("Level10")
        end
    end,

    OnStart = function (self)
        self.focus = self.actor:GetComponent("ButtonFocus")
        Event.Subscribe("ButtonPress", self, self._OnPress)
    end

}